Unable to build shogun's python interface

Hi, I have been trying to build shogun's python interface from source code. But, I always get the following error message (I have turned on verbose for the make command. Hence the lengthy messages):
```
[ 74%] Built target python_doxy2swig
make -f src/interfaces/python/CMakeFiles/interface_python_src.dir/build.make src/interfaces/python/CMakeFiles/interface_python_src.dir/depend
make[2]: Entering directory '/media/Rukman/IIT_B/gumption/shogun/build_python'
cd /media/Rukman/IIT_B/gumption/shogun/build_python && /usr/bin/cmake -E cmake_depends "Unix Makefiles" /media/Rukman/IIT_B/gumption/shogun /media/Rukman/IIT_B/gumption/shogun/src/interfaces/python /media/Rukman/IIT_B/gumption/shogun/build_python /media/Rukman/IIT_B/gumption/shogun/build_python/src/interfaces/python /media/Rukman/IIT_B/gumption/shogun/build_python/src/interfaces/python/CMakeFiles/interface_python_src.dir/DependInfo.cmake --color=
make[2]: Leaving directory '/media/Rukman/IIT_B/gumption/shogun/build_python'
make -f src/interfaces/python/CMakeFiles/interface_python_src.dir/build.make src/interfaces/python/CMakeFiles/interface_python_src.dir/build
make[2]: Entering directory '/media/Rukman/IIT_B/gumption/shogun/build_python'
[ 74%] copying SWIG files
make[2]: Leaving directory '/media/Rukman/IIT_B/gumption/shogun/build_python'
[ 74%] Built target interface_python_src
make -f src/interfaces/python/CMakeFiles/_interface_python.dir/build.make src/interfaces/python/CMakeFiles/_interface_python.dir/depend
make[2]: Entering directory '/media/Rukman/IIT_B/gumption/shogun/build_python'
[ 75%] Swig source
cd /media/Rukman/IIT_B/gumption/shogun/build_python/src/interfaces/python && /usr/bin/cmake -E make_directory /media/Rukman/IIT_B/gumption/shogun/build_python/src/interfaces/python
cd /media/Rukman/IIT_B/gumption/shogun/build_python/src/interfaces/python && /usr/bin/swig3.0  -python -builtin -modern -modernargs -threads -w473 -w454 -w312 -w325 -fvirtual -outdir /media/Rukman/IIT_B/gumption/shogun/build_python/src/interfaces/python -c++ -I/usr/include/python2.7 -I/usr/lib/python2.7/dist-packages/numpy/core/include -I/media/Rukman/IIT_B/gumption/shogun/src -I/media/Rukman/IIT_B/gumption/shogun/build_python/src -I -I/media/Rukman/IIT_B/gumption/shogun/src/gpl -I/media/Rukman/IIT_B/gumption/shogun/build_python/StanMath/src/StanMath -I -I/media/Rukman/IIT_B/gumption/shogun/build_python/StanMath/src/StanMath/lib/boost_1.66.0 -I -I/media/Rukman/IIT_B/gumption/shogun/build_python/StanMath/src/StanMath/lib/sundials_3.1.0/include -I -I/media/Rukman/IIT_B/gumption/shogun/build_python/StanMath/src/StanMath/lib/eigen_3.3.3 -I -I/media/Rukman/IIT_B/gumption/shogun/build_python/src/shogun/third_party -I -I/media/Rukman/IIT_B/gumption/shogun/build_python/src/shogun/third_party/variant -I -I/usr/include -I/usr/include/json-c -I/usr/include/libxml2 -I/usr/include/hdf5/serial -o /media/Rukman/IIT_B/gumption/shogun/build_python/src/interfaces/python/shogunPYTHON_wrap.cxx /media/Rukman/IIT_B/gumption/shogun/build_python/src/interfaces/python/shogun.i
shogun_doxygen.i:612: Error: Syntax error in input(1).
src/interfaces/python/CMakeFiles/_interface_python.dir/build.make:64: recipe for target 'src/interfaces/python/shogunPYTHON_wrap.cxx' failed
make[2]: *** [src/interfaces/python/shogunPYTHON_wrap.cxx] Error 1
make[2]: Leaving directory '/media/Rukman/IIT_B/gumption/shogun/build_python'
CMakeFiles/Makefile2:475: recipe for target 'src/interfaces/python/CMakeFiles/_interface_python.dir/all' failed
make[1]: *** [src/interfaces/python/CMakeFiles/_interface_python.dir/all] Error 2
make[1]: Leaving directory '/media/Rukman/IIT_B/gumption/shogun/build_python'
Makefile:154: recipe for target 'all' failed
make: *** [all] Error 2
```
I have looked at the 612 line of the `shogun_doxygen.i` file but I am not sure I understand the error generated. I am also posting the [shogun_doxygen.i][1] file as a gist.

Thanks in advance for the help :)

[1]: https://gist.github.com/braceletboy/c2685d8c5bb768a4544d172e9199ab5f

@gf712 Thanks! This problem was solved. But I met a new error when I run 'make' command, here is the error message about new problem:

[ 91%] Built target cpp-kernel-auc
[ 91%] Linking CXX executable distance/cpp-distance-manhattan
[ 91%] Built target cpp-distance-manhattan
[ 91%] Linking CXX executable kernel/cpp-kernel-anova
[ 91%] Built target cpp-kernel-anova
[ 91%] Linking CXX executable kernel/cpp-kernel-chi2
[ 91%] Built target cpp-kernel-chi2
[ 91%] Linking CXX executable kernel/cpp-kernel-log
[ 91%] Built target cpp-kernel-log
[ 91%] Linking CXX executable neural_nets/cpp-neural_nets-feedforward_net_classification
[ 91%] Built target cpp-neural_nets-feedforward_net_classification
[ 91%] Linking CXX executable evaluation/cpp-evaluation-cross_validation
[ 91%] Built target cpp-evaluation-cross_validation
[ 91%] Linking CXX executable meta_api/cpp-meta_api-calls
[ 91%] Built target cpp-meta_api-calls
[ 91%] Building CXX object examples/meta/cpp/CMakeFiles/cpp-meta_api-literals.dir/meta_api/literals.cpp.o
/home/zhangjing/packages/shogun/build/UILD_META_EXAMPLES=ON/examples/meta/cpp/meta_api/literals.cpp:5:25: error: ‘io’ is not a namespace-name
 using namespace shogun::io;
                         ^~
/home/zhangjing/packages/shogun/build/UILD_META_EXAMPLES=ON/examples/meta/cpp/meta_api/literals.cpp:5:27: error: expected namespace-name before ‘;’ token
 using namespace shogun::io;
                           ^
examples/meta/cpp/CMakeFiles/cpp-meta_api-literals.dir/build.make:62: recipe for target 'examples/meta/cpp/CMakeFiles/cpp-meta_api-literals.dir/meta_api/literals.cpp.o' failed
make[2]: *** [examples/meta/cpp/CMakeFiles/cpp-meta_api-literals.dir/meta_api/literals.cpp.o] Error 1
CMakeFiles/Makefile2:5592: recipe for target 'examples/meta/cpp/CMakeFiles/cpp-meta_api-literals.dir/all' failed
make[1]: *** [examples/meta/cpp/CMakeFiles/cpp-meta_api-literals.dir/all] Error 2
Makefile:151: recipe for target 'all' failed
make: *** [all] Error 2

Do you have any idea about this question? I'm sorry to disturb you again but I have no idea about this.

--- a/src/shogun/lib/any.h
+++ b/src/shogun/lib/any.h
@@ -383,6 +383,7 @@ namespace shogun
 		template <>
 		bool compare_impl_eq(const complex128_t& lhs, const complex128_t& rhs);
 
+#ifndef DOXYGEN_SHOULD_SKIP_THIS
 		template <typename T, typename _ = void>
 		struct has_special_compare : std::false_type
 		{
@@ -419,6 +420,7 @@ namespace shogun
 		           std::declval<T>()))>> : public std::true_type
 		{
 		};
+#endif // DOXYGEN_SHOULD_SKIP_THIS
 
 		template <class T>
 		constexpr T& mutable_value_of(void** ptr)

--- a/src/shogun/util/traits.h
+++ b/src/shogun/util/traits.h
@@ -7,6 +7,8 @@ namespace shogun
 {
 	namespace traits
 	{
+		#ifndef DOXYGEN_SHOULD_SKIP_THIS
+
 		template<typename... Ts>
 		struct try_to_declare {};
 
@@ -147,7 +149,8 @@ namespace shogun
 
 		template<typename T>
 		using returns_void = std::is_same<typename T::result_type, void>;
-
+		
+		#endif // DOXYGEN_SHOULD_SKIP_THIS
 	} // namespace traits
 } // namespace shogun
 #endif

